featurebyte/config.py

```diff
@@ -347,14 +369,15 @@ def _parse_config(self, path: Path) -> None:
             self.profiles = ProfileList(profiles=profile_settings).profiles
 
         if self.profiles:
-            # use first profile as fallback
-            self.profile = self.profiles[0]
+            profile_map = {profile.name: profile for profile in self.profiles}
+            default_profile = self.settings.pop("default_profile", None)
             selected_profile_name = os.environ.get("FEATUREBYTE_PROFILE")
             if selected_profile_name:
-                for profile in self.profiles:
-                    if profile.name == selected_profile_name:
-                        self.profile = profile
-                        break
+                self._profile = profile_map.get(
+                    selected_profile_name, profile_map.get(default_profile)
+                )
```

Comment:
Curious why don't we throw error to inform user that the specified profile does not exist?